qtaskbarcontrol_x11.cpp

```diff
@@ -55,26 +45,16 @@ void QX11TaskbarControl::setCounter(bool counterVisible, int counter)
 
 void QX11TaskbarControl::sendMessage(const QVariantMap &params)
 {
-	if(_desktopFile.isEmpty()) {
-		qWarning() << "You need to set the" << QTaskbarControl::LinuxDesktopFile << "attribute before you can use QTaskbarControl!";
+	if(QGuiApplication::desktopFileName().isEmpty()) {
+		qWarning() << "You need to set the desktop file name before you can use QTaskbarControl!";
 		return;
 	}
 
 	auto message = QDBusMessage::createSignal(QStringLiteral("/de/skycoder42/QTaskbarProgress"),
 											  QStringLiteral("com.canonical.Unity.LauncherEntry"),
 											  QStringLiteral("Update"));
 
-	message << QString{QStringLiteral("application://") + _desktopFile}
+	message << QStringLiteral("application://") + QGuiApplication::desktopFileName()
```

Comment:
Have you tested this with `QT_USE_QSTRINGBUILDER` defined? If I remember correctly, I had to add the `QString` part to force a conversion in such cases.

Reply:
Just checked with `QT_USE_QSTRINGBUILDER` it works. `QStringLiteral` and `QGuiApplication::desktopFileName()` return a `QString` and merge into one, which is equivalent to `QString {... + ...}`

Reply:
Not quite, If this define exists, then `QString + QString` evaluates to `QStringBuilder<...>`, which is implicitly convertible to `QString`. This can cause problems sometimes, but apparently works correctly with the stream operator. You can leave it as is.